### ISSUE TITLE
Considered Netlify?

### ISSUE THREAD

**Issue by kaushalmodi** [zzamboni/ng#2]
Hello,

I see that you manually commit the HTML files in the "Regen" commits. So I was wondering if you considered using Netlify to deploy your site (*I have zero affiliation with that website*).

Once you create a Netlify account, you can have it track any git repo and set the command (`hugo` in this case) to run on it to generate your site. It eliminates the HTML commit noise (like Gitlab CI, but better: free HTTPS, etc.).

For example, this is all I need to configure for Netlify based deployment of the ox-hugo doc site:

![image](https://user-images.githubusercontent.com/3578197/43729700-19055316-9977-11e8-841e-f3bbfc97c661.png)

- `doc_md` rule generates the Markdown files from Org using `ox-hugo`
- `hugo_doc HUGO=..` runs `hugo` but using my personally built `hugo` binary

So the end result is: 
1. Commit Org file
2. Site updated in seconds :D

**Comment by zzamboni** [comment]
Hey @kaushalmodi - this is so weird - precisely **today** I've been playing with Netlify! Here's the screenshot from my other tab right now, I kid you not:

![image](https://user-images.githubusercontent.com/32876/43736039-988fd12a-99bc-11e8-88c9-45507fbed45b.png)

Yes, I'm still fighting with some weird filenames that are included in my website because I'm carrying cruft from old incarnations, back almost 20 years. So still a bit of cleanup to do, but it looks very promising.

Thanks so much for the configuration suggestion - I was still thinking of exporting from my org file by hand and have Netlify just run `hugo`, but I see that you can automate also the ox-hugo export - very cool!

I personally find the need to export my org file locally and run `hugo server` to check the result, for my own piece of mind. Also, I have a lot of old stuff in `content`, so I cannot simply ignore the full directory in git. Any ideas for dealing with this?

**Comment by kaushalmodi** [comment]
> this is so weird - precisely today I've been playing with Netlify! Here's the screenshot from my other tab right now, I kid you not

That's really odd! I was going to mention Netlify at the time your site was added to the ox-hugo Real World Examples page. It's just today that I thought of commenting about it.

> Yes, I'm still fighting with some weird filenames that are included in my website

Yes, that's some interesting file name :)

> I was still thinking of exporting from my org file by hand and have Netlify just run hugo

I do the same for my personal site. The exporting from Org over Netlify works for the doc site works as I don't care about preserving "Last modified" times there, and also the site has just a handful of pages.

> I personally find the need to export my org file locally and run hugo server to check the result, for my own piece of mind.

That makes sense. 

> Also, I have a lot of old stuff in content, so I cannot simply ignore the full directory in git. Any ideas for dealing with this?

Having old committed content works alright with ox-hugo, as it simply adds/overwrites whatever new Org content you export. So I don't understand why old content is a problem.. unless if you mean ignoring the old content with the above mentioned file names issue. 

In that case, I'd suggest removing such problematic content for test purpose in a branch and making using of the [Netlify Deploy contexts][1]. Example in `netlify.toml` [[ref][2]]:

```toml
[context.deploy-preview]
  command = "HUGO_BASEURL=$DEPLOY_PRIME_URL/ hugo"
```

[1]: https://www.netlify.com/docs/continuous-deployment/#deploy-contexts
[2]: https://gitlab.com/kaushalmodi/kaushalmodi.gitlab.io/blob/f2473feac5e84a72925935aee6eb768f8ba8c7cc/netlify.toml#L34-46

**Comment by zzamboni** [comment]
First successful Netlify publish! https://determined-johnson-8e3a34.netlify.com/ Next up: checking that everything seems to work, and then reconfiguring my domain to point to it.

> I do the same for my personal site. The exporting from Org over Netlify works for the doc site works as I don't care about preserving "Last modified" times there, and also the site has just a handful of pages.

Makes sense. I think I might do that for my book's website (http://cf-learn.info/).

> Having old committed content works alright with ox-hugo, as it simply adds/overwrites whatever new Org content you export. So I don't understand why old content is a problem.. unless if you mean ignoring the old content with the above mentioned file names issue.

I was thinking of a case in which:

a) You have old stuff under `content/` which needs to be checked in;
b) You have new stuff in ox-hugo which gets exported automatically to `content/`.

In this case, only some things under `content/` would need to be ignored in git so they are not stored twice. I guess one solution would be to store the "prior stuff" somewhere else, and only copy it to `content/` as part of the build process, before exporting from ox-hugo. Anyhow, something to think about :)

Thanks for the tips, and as usual, thanks for ox-hugo and for your Emacs expertise!

**Comment by kaushalmodi** [comment]
> First successful Netlify publish!

Yay!

> I was thinking of a case in which ..

It's still not clear.. Are you talking about the case where Org -> Md exporting using ox-hugo happens on Netlify?

**If yes**:

- `ox-hugo` will simply add/overwrite already committed stuff in `content/`.

**If no**:

- You commit everything you want Hugo to process in `content/` manually.

So I did not understand what needs to be ignored in `content/`. May be an example can help me understand this more.

> Thanks for the tips,

You are very much welcome! :D

**Comment by zzamboni** [comment]
> So I did not understand what needs to be ignored in content/. May be an example can help me understand this more.

Probably I'm getting confused myself. Let me play with it a bit, I think that will clarify in my mind if there's really a question worth asking :)

Quick question: are you also using Netlify DNS?

At the moment I'm waiting for the DNS changes in my current registrar to propagate so that my domain is served by Netlify - hopefully by tomorrow everything should be propagated.

**Comment by kaushalmodi** [comment]
> are you also using Netlify DNS?

Yes.

> At the moment I'm waiting for the DNS changes in my current registrar to propagate

Yeah, that takes about an hour.

**Comment by zzamboni** [comment]
Success! My website https://zzamboni.org/ is now fully hosted on Netlify. The process was overall very easy, although I'd like to note some particularities for posterity:

- Since my site carries a lot of old content, I had to clean up a few filenames which contained `#`, `?` or some other characters not allowed by Netlify.
- I decided to switch to Netlify DNS. The DNS switchover took longer than expected due to a failure in the DNS server of my [domain provider](http://hoststar.ch/) - after giving it a few hours, a call to support led to a quick realization and fix of the problem.
- I had initially some strange problems with infinite redirection loops in some of my pages. Eventually I realized that they were caused by Hugo aliases, which don't work properly in Netlify. I switched to implementing aliases using Netlify redirects. [This post](https://discourse.gohugo.io/t/solved-posts-in-netlify-fail-to-load/9177) in the Hugo forum was my first pointer in the right direction, and I later found [this blog post](https://www.netlify.com/docs/build-settings/) with more details about automatically generating the Netlify `_redirects` file with Hugo.

@kaushalmodi thanks again for your ping and for the tips!
